### PR TITLE
Add AppVeyor as a CI on Window

### PR DESCRIPTION
**Pull request by MasseGuillaume**
https://github.com/sbt/sbt/issues/3908

### CODE REVIEW

**Comment by MasseGuillaume** [comment]
Progress:
https://ci.appveyor.com/project/scalacenter/scalafix/build/26/job/gb6tyfttsrdsg25o#L6885

```
[error] Failed tests:
[error] 	scalafix.tests.reflect.ScalafixReflectTests
[error] 	scalafix.tests.cli.CliGitDiffTests
[error] 	scalafix.tests.cli.CliSemanticTests
[error] 	scalafix.tests.core.PatchSuite
[error] 	scalafix.tests.reflect.ToolClasspathTests
```

**Comment by MasseGuillaume** [comment]
Related:
* https://github.com/scalacenter/scalafix/issues/583
* https://github.com/scalacenter/scalafix/issues/547
* https://github.com/scalacenter/scalafix/issues/364

File: project/ScalafixBuild.scala [hunk 1]
```diff
@@ -181,21 +184,41 @@ object ScalafixBuild extends AutoPlugin with GhpagesKeys {
...
         s
     },
     commands += Command.command("ci-fast-211") { s =>
       "scalafix211/test" ::
         s
     },
+    commands += Command.command("ci-sbt-windows") { s =>
+      // scripted tests don't work in sbt 1.0 yet because we run Sbt1
+      "scalafix-sbt/publishLocal" ::
+        testSkipWindows("scalafix-sbt") ::
```
Comment (olafurpg) [inline project/ScalafixBuild.scala]: Do we skip scripted tests on windows?

File: project/ScalafixBuild.scala [hunk 2]
```diff
@@ -181,21 +184,41 @@ object ScalafixBuild extends AutoPlugin with GhpagesKeys {
...
+    commands += Command.command("ci-sbt013-windows") { s =>
+      "scalafix-sbt013/scripted" ::
+        testSkipWindows("scalafix-sbt013") ::
+        s
+    },
+    commands += Command.command("ci-fast-212-windows") { s =>
+      testSkipWindows("scalafix") ::
+        s
+    },
+    commands += Command.command("ci-fast-211-windows") { s =>
```
Comment (olafurpg) [inline project/ScalafixBuild.scala]: Should we maybe only run 2.12 + JVM on windows?

File: appveyor.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+environment:
+  JAVA_HOME: C:\Program Files\Java\jdk1.8.0
+  APPVEYOR_SAVE_CACHE_ON_ERROR: true
+  APPVEYOR_CACHE_ENTRY_ZIP_ARGS: "-t7z -m0=lzma -mx=9"
+
+  matrix:
+    - CI_TEST: ci-fast-212
+    # - CI_TEST: ci-fast-211 to shave some time 
+    - CI_TEST: ci-sbt
```
Comment (olafurpg) [inline appveyor.yml]: Can we skip this? The sbt 1 entry doesn't run any tests iirc

File: project/ScalafixBuild.scala (hunk 1 above)
Comment (MasseGuillaume) [reply project/ScalafixBuild.scala]: no this is scalafix-sbt/test

**Comment by olafurpg** [comment]
Seems like we'll have to add a method overload for `checkDiff` in testkit to please mima https://travis-ci.org/scalacenter/scalafix/jobs/335659555#L799 we can keep the original signature that forwards to another `checkDiff` that has at least one non-variadic `Tag`

**Comment by olafurpg** [comment]
The appveyor failure is reproducing #583 https://ci.appveyor.com/project/scalacenter/scalafix/build/33/job/ddm4qbjd2hqxyfao#L5494 Can we can disable it for now and I will try and fix it in a followup PR

**Comment by fommil** [comment]
btw take a look at any of my appveyor setups (particularly ensime-server or ensime-sbt) to see how to get symlinks to work. Look for magical git commands in the preamble.

**Comment by MasseGuillaume** [comment]
@fommil the symlink was not used, I just removed it.

**Pull request merged**